src/utils.tsx

```diff
@@ -1,13 +1,28 @@
-import { isFunction, isString } from "lodash"
+import cons from "consolidate"
+import { isFunction } from "lodash"
+import path from "path"
+import { Engine, StitchConfig } from "./index"
 import { Block } from "./render"
 
-export function isTemplate(block: Block): boolean {
-  if (!isString(block)) {
-    return false
-  }
-  const BLACKLIST = [".js", ".jsx", ".ts", ".tsx"]
-  const found = BLACKLIST.some(extension => block.includes(extension))
-  return !found
+const BLACKLIST = [".js", ".jsx", ".ts", ".tsx"]
+
+// TODO: Could stat the FS to make sure this is truly an existing template.
+export function isTemplate(
+  filePath: string,
+  engines?: StitchConfig["engines"]
+): boolean {
+  return (
+    !BLACKLIST.includes(path.extname(filePath)) &&
+    !!getCompileFn(filePath, engines)
```

Comment:
So now we consider it a template if there is an engine for the extension and it's not blacklisted.

Reply:
I'm now thinking that we can probably get rid of the blacklisting as I don’t think there would ever be an engine for those blacklisted extensions, no?

Reply:
Updated